zcash_client_sqlite/src/wallet/init/migrations/v_transactions_net.rs

```diff
@@ -0,0 +1,498 @@
+            "DROP VIEW v_transactions;
+            CREATE VIEW v_transactions AS
+            WITH
+            notes AS (
+                SELECT received_notes.account        AS account_id,
+                       received_notes.tx             AS id_tx,
+                       received_notes.value          AS value,
+                       CASE
+                            WHEN received_notes.is_change THEN 1
+                            ELSE 0
+                       END AS is_change,
+                       CASE
+                            WHEN received_notes.is_change THEN 0
+                            ELSE 1
+                       END AS received_count,
+                       CASE
+                           WHEN received_notes.memo IS NULL THEN 0
+                           ELSE 1
+                       END AS memo_present
+                FROM   received_notes
+                UNION
+                SELECT received_notes.account        AS account_id,
+                       received_notes.spent          AS id_tx,
+                       -received_notes.value         AS value,
+                       0                             AS is_change,
+                       0                             AS received_count,
+                       0                             AS memo_present
+                FROM   received_notes
+                WHERE  received_notes.spent IS NOT NULL
+            ),
+            sent_note_counts AS (
+                SELECT sent_notes.from_account AS account_id,
+                       sent_notes.tx AS id_tx,
+                       COUNT(DISTINCT sent_notes.id_note) as sent_notes,
+                       SUM(
+                         CASE
+                             WHEN sent_notes.memo IS NULL THEN 0
+                             ELSE 1
+                         END
+                       ) AS memo_count
+                FROM sent_notes
+                LEFT JOIN received_notes
+                          ON (sent_notes.tx, sent_notes.output_pool, sent_notes.output_index) =
+                             (received_notes.tx, 2, received_notes.output_index)
+                WHERE  received_notes.is_change IS NULL
+                   OR  received_notes.is_change = 0
+                GROUP BY account_id, id_tx
+            ),
+            blocks_max_height AS (
+                SELECT MAX(blocks.height) as max_height FROM blocks
+            )
+            SELECT notes.account_id                  AS account_id,
+                   transactions.id_tx                AS id_tx,
+                   transactions.block                AS mined_height,
+                   transactions.tx_index             AS tx_index,
+                   transactions.txid                 AS txid,
+                   transactions.expiry_height        AS expiry_height,
+                   transactions.raw                  AS raw,
+                   SUM(notes.value)                  AS net_transfer,
```

Comment:
@nuttycom I've tested it and I have a doubt. It could be either an error in the query or a misunderstanding.

When I send 20000 Zats I see that the mined transaction is read back from the chain with and the value present in `net_transfer` is -21000 and `fee_paid` of 1000.

I would expect that `net_transfer` is the value of the transaction minus all deductions (in this case the fee), and when I inspect the value I find the "gross" value, which is the whole value transferred, which I consider it to be the value indicated by user intent to be transferred to the recipient, plus the value of the fee (which in this case is 1000 but if could be something else less noticeable).

I see a couple of paths here:
a) this is intentional. The value property of a transaction in the SDK will show the sum total value transferred to recipients and the naming and to show the value transferred to recipients without the fee the wallet must add the fee to the `net_transfer` column

b) there's a mistake. The value property of a transaction in the SDK will show the value transferred to recipients and to show the whole value of the transaction, wallets must add the fee to the value present  in the `net_transfer` column. 

Also, fee value for received transactions is zero

Reply:
@pacu your answer is (a), this is intentional. The reason for this is, consider this scenario:

Your wallet recovers a transaction that:
* Spends a 500 ZAT note belonging to account `0`
* Spends a 20500 ZAT note belonging to account `1`
* Has an output of 20000 ZAT that goes to an external address.

This transaction will result in two entries in `v_transactions`, one associated with each account. The problem is, where to associate the fee? 

Consider further: the ECC wallet only attempts recovery for account `0`. Now, you show as one of "your" wallet's transactions... what? The answer we have here is a transfer of `-500`, a fee of 1000, and 1 sent output.

Reply:
there is no emoji to reaction to express my "mindblown-ness" 😅

Should wallets display the `net_transfer` straightforward in any case then? or shall they add or substract  the fee?

Reply:
I think that wallets should show `net_transfer` directly.

Reply:
> The value property of a transaction in the SDK will show the sum total value transferred to recipients and the naming and to show the value transferred to recipients without the fee the wallet must add the fee to the `net_transfer` column

Yes. Framing this another way, the `net_transfer` column is the total amount that the balance of the account increases or decreases by.

Reply:
Or maybe `account_net_value`?

Reply:
I think the main naming conflict here is that the "net value" of something is the total value minus all possible deductions which is not quite what's going on here.